script.js

```diff
@@ -0,0 +1,289 @@
+                    </div>
+                </div>
+                <div class="form-group">
+                    <label for="date">Due Date</label>
+                    <input type="date" class="form-control" id="date-edit">
+                </div>
+                <button type="submit" class="btn btn-primary">Update</button>
+            `)  
+            let activity = ['makan', 'dinner', 'nongkrong', 'hangout']
+            for (let i = 0; i < activity.length; i++) {
+                if(data.Todo.title.toLowerCase() == activity[i]){
+                    for (let j = 0; j < data.recommendations.length; j++) {
+                        $('#table-recom-by-Id').append(`
+                            <tr>
+                                <td>${data.recommendations[j].title}</td>
+                                <td>${data.recommendations[j].description}</td>
+                                <td><a href="${data.recommendations[j].url}">${data.recommendations[i].url}</a></td>
+                            </tr>
+                        `)
+                    }
+                } 
+            }
+        })
+        .fail(err => console.log(err))
+    });
+}
+
+function editTodo(event){
+    event.preventDefault()
+    let updatePayload = {
+        title: $('#title-edit').val(),
+        description: $('#description-edit').val(),
+        status: $("input[name='customRadio']:checked").val(),
+        due_date: $('#date-edit').val()
+    }
+    $('#table-todo-by-Id tr').each(function() {
+        let id = $(this).find(".todo-edit-id").html();
+        $.ajax({
+            method: 'put',
+            url: `${baseurl}/todos/${id}`,
+            headers: {
+                token: localStorage.token
+            },
+            data: updatePayload
+        })
+            .done(data => {
+                fetchTodos()
+                $('#edit-page').hide()
+                $('#signin-page').hide()
+                $('#home-page').show()
+                
+            })
+            .fail(err => {
+                console.log(err.responsesJSON.error)
+            })
+
+            .always(() => {
+                $('#title-edit').val('')
+                $('#description-edit').val('')
+            })
+    });
+}
+
+function deleteTodo(event){
+    event.preventDefault()
+    $('#table-todo-by-Id tr').each(function(){
+        let id = $(this).find(".todo-edit-id").html()
+        $.ajax({
+            method: 'delete',
+            url: `${baseurl}/todos/${id}`,
+            headers: {
+                token: localStorage.token
+            }
+        })
+            .done(() => {
+                fetchTodos()
+                $('#edit-page').hide()
+                $('#signin-page').hide()
+                $('#home-page').show()
+            })  
+            .fail(err => {
+                console.log(err.responseJSON.error)
+            })    
+    })
+}       
+
+function onSignIn(googleUser) {
```

Comment:
google sign in masih gagal ya, tampil pesan validation error

Reply:
mas di laptopku  google sign in kok berhasil ya